Update guidance information for a11ydialog

joesnellpdx opened the pull request:
### Description of the Change

We recommend a 3rd party implementation for dialogs and modals - [A11yDialog](https://github.com/KittyGiraudel/a11y-dialog). This package introduced a breaking release v7. For this MR, we are refining the description and information, adding migration guidance, and altering the implementation to match the new release. 

### Alternate Designs

N/A

### Benefits

Guidance aligns with the new release of A11yDialog

### Possible Drawbacks

None.

### Verification Process

Tested package usage and reviewed [demo](https://codesandbox.io/s/a11y-dialog-v7-pnwqu).

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

joesnellpdx committed:
Update guidance information for a11ydialog

smy315 commented on docs/_component/modal/index.md:
```diff
@@ -1,7 +1,7 @@
 layout: default
-seo-title: Modal
+seo-title: Modal / Dialog
 description: A third party plugin we use to produce an accessible modal dialog component.
```
Since you updated all the titles to include "/ Dialog", make the description consistent as well and update `modal dialog` to `modal / dialog`?

smy315 commented on docs/_component/modal/component.html:
```diff
@@ -1,11 +1,20 @@
     </button>
-    <h1 id="dialog-title">Dialog Title</h1>
+    <!-- 5. The dialog title -->
+    <h1 id="your-dialog-title-id">Your dialog title</h1>
```
I know this existed before but update h1 to h2. This code sample will most likely be copied and pasted as boilerplate and there will most likely be h1 already on a page.

Actually while doing Accessibility Audits in the past, I have often caught duplicate h1 issues because of it being used on modals while h1 was already used for site or page titles on a page.

smy315 commented:
@joesnellpdx left two minor comments on PR but other than that, I think this one is good to merge!

joesnellpdx committed:
Merge branch 'develop' into feature/44-component-modal-allydialog-7

joesnellpdx committed:
Update text and semantics
- Change page description to contain modal / dialog
- Change markup guidance for dialog title to be an h2 rather than an h1
